### PR TITLE
[Fix] mzTab single proteins

### PR DESCRIPTION
**Pull request by jpfeuffer**
Somehow they were previously just ignored.

### CODE REVIEW

**Comment by jpfeuffer** [comment]
Since I changed some indents my editor auto-formatted (e.g. split) some lines.

File: src/openms/source/FORMAT/MzTab.cpp [hunk 1]
```diff
@@ -1877,20 +1881,20 @@ namespace OpenMS
         if (!ms_run_in_data.empty())
         {
           // prepend file:// if not there yet
-          for (const String& s : ms_run_in_data)
+          for (const String &s : ms_run_in_data)
```
Comment (timosachsenberg) [inline src/openms/source/FORMAT/MzTab.cpp:1884]: do we prefer String &s over String& s ?

Comment (jpfeuffer) [reply src/openms/source/FORMAT/MzTab.cpp:1884]: ah good catch. have to change my editor settings

File: src/openms/source/FORMAT/MzTab.cpp [hunk 2]
```diff
@@ -2179,15 +2128,68 @@ namespace OpenMS
...
+        // map general protein groups to the contained proteins
         // We only report quantitative data for indistinguishable groups (which may be composed of single proteins).
         // We skip the more extensive reporting of general groups with complex shared peptide relations.
-        std::vector<ProteinIdentification::ProteinGroup> protein_groups2;
+        std::vector<ProteinIdentification::ProteinGroup> empty_protein_groups;
+        auto protein_groups = std::cref(empty_protein_groups);
         if (quant_study_variables == 0)
         {
-          protein_groups2 = it->getProteinGroups();
+          protein_groups = std::cref(it->getProteinGroups());
```
Comment (timosachsenberg) [inline src/openms/source/FORMAT/MzTab.cpp:2168]: is there a difference to const std::vector<ProteinIdentification::ProteinGroup>& protein_groups2 = quant_study_variables_ == 0 ? pid.getProteinGroups() : std::vector<ProteinIdentification::ProteinGroup>();
?

**Comment by timosachsenberg** [comment]
integrated into streaming PR

**Pull request closed without merging**